src/hotspot/share/memory/universe.cpp

```diff
@@ -580,6 +589,21 @@ oop Universe::out_of_memory_error_realloc_objects() {
   return gen_out_of_memory_error(out_of_memory_errors()->obj_at(_oom_realloc_objects));
 }
 
+bool Universe::is_out_of_memory_error(oop obj, const char* msg) {
+  assert(msg != NULL, "sanity check");
+  oop msg_oop = java_lang_Throwable::message(obj);
+  const char * msg_str = java_lang_String::as_utf8_string(msg_oop);
+  return strcmp(msg, msg_str) == 0;
+}
```

Comment:
Yes, I agree with David about strcmp.  Could you add this instead (it was trying to compare the oops for equality, and unintentionally generated a new one).

oop Universe::is_out_of_memory_error_metaspace(oop exception) {
  return out_of_memory_errors()->obj_at(_oom_metaspace)) == exception;
}

Reply:
Ok, I see how this isn't right, but I still don't like the strcmp.  Can the string be interned and checked for equality?  Otherwise you also need a ResourceMark here.

Reply:
This would still need a ResourceMark, since java_lang_String::as_utf8_string() allocates into a resource allocated array.